### PR TITLE
Update destructuring docs

### PR DESCRIPTION
**Pull request by kyldvs**
Just some minor edits. I added a little bit more organization and removed some of the more advanced features.

I'm thinking we might want to consolidate the advanced usages of destructuring into an article under "advanced features" section. I think this could also include the advanced versions involving variants and get them all in one place

### CODE REVIEW

File: docs/destructuring.md [hunk 1]
```diff
@@ -2,59 +2,65 @@
...
-The following binds variables: `ten = 10`, `twenty = 20`
+## Tuples
 
 ```reason
-let someInts = (10, 20);
-let (ten, twenty) = someInts;
-```
+let pair = (1, "Hello");
 
-The following binds variables: `name = "Guy"`, `age = 30`
```
Comment (jordwalke) [inline docs/destructuring.md:16]: I kind of prefer single line comments for inline code example comments. For some reason it looks a little cleaner/familiar.

File: docs/destructuring.md [hunk 2]
```diff
@@ -2,59 +2,65 @@
...
+let {name, age} = alice;
+
+/* Prints: Alice */
+print_endline(name);
 ```
 
-**Keep reading the section, pattern matching, for a crazier form of destructuring**!
+### Renaming Record Fields
+
+While destructuring it is possible to bind to variables different than the field
```
Comment (jordwalke) [inline docs/destructuring.md:44]: A little more accurate:

> While destructuring record fields, it is possible to bind variable names that are different than the record field names.

File: docs/destructuring.md [hunk 3]
```diff
@@ -2,59 +2,65 @@
...
-Destructuring can make your code much more concise without requiring you to name intermediate variables. Do use them! But don't abuse them and make your code overly nested & terse.
+```reason
+/* Without destructuring */
+let hello = (person) => {
+  print_endline("Hello " ++ person.name);
+};
 
-If you're destructuring a record or a variant whose definition isn't in the current file, you need to explicitly annotate it. See [here](record.md#record-needs-an-explicit-definition) and [here](variant.md#variant-needs-an-explicit-definition).
+/* With destructuring */
+let hello = ({name}) => {
```
Comment (jordwalke) [inline docs/destructuring.md:63]: Could also add an example of your aliasing inside a function argument. `({name: theName}) => `

File: docs/destructuring.md [hunk 4]
```diff
@@ -2,59 +2,65 @@
...
+## Function Arguments
 
-Destructuring can make your code much more concise without requiring you to name intermediate variables. Do use them! But don't abuse them and make your code overly nested & terse.
+```reason
+/* Without destructuring */
+let hello = (person) => {
+  print_endline("Hello " ++ person.name);
+};
 
-If you're destructuring a record or a variant whose definition isn't in the current file, you need to explicitly annotate it. See [here](record.md#record-needs-an-explicit-definition) and [here](variant.md#variant-needs-an-explicit-definition).
```
Comment (jordwalke) [inline docs/destructuring.md:60]: I think this note was pretty helpful. Maybe include it?

File: docs/destructuring.md (hunk 1 above)
Comment (kyldvs) [reply docs/destructuring.md:16]: I do too, but the syntax highlighting isn't supported. I'm happy to go back and update them all if support is added

Comment (kyldvs) [reply docs/destructuring.md:16]: Another consideration is that sketch doesn't support single line comments. If we want examples to be copy/pastable in there then we should stick to `/* */` style comments or get support added in sketch.

**Pull request merged**